databases/week2/homework2.txt

```diff
@@ -0,0 +1,177 @@
+CHARLES MAINA DATABASE WEEK 2 HOMEWORK, CLASS 13.
+PART1.
+USE hyflesson2;
+
+--Add a task with these attributes: title, description, created, updated, due_date, status_id, user_id
+
+ INSERT INTO task
+ (title, 
+description, 
+created,
+ updated, 
+ due_date, 
+status_id)
+ VALUES (
+'chess',
+ 'play with queen and pawns, bishop is defender', 
+'2020-05-15',
+'2020-05-20',
+'2020-05-29',
+3);
+ 
+--Change the title of a task
+ UPDATE task
+ SET title= 'cards'
+ WHERE title= 'chess';
+ 
+--Change a task due date
+ UPDATE task
+ SET due_date= '2020-06-01'
+ WHERE due_date= '2020-05-29';
+ 
+--Change a task status
+ UPDATE task
+ SET status_id= 2
+ WHERE status_id= 3;
+
+--mark status as done
+UPDATE task
+ SET status_id= 3
+ WHERE status_id= 2;
+
+--Delete task.
+DELETE task
+ WHERE title = 'cards';
+
+
+PART 2.
+SCHOOL DATABASE.
+/*Create a new database containing the following tables:
+•	Class: with the columns: id, name, begins (date), ends (date)
+•	Student: with the columns: id, name, email, phone, class_id (foreign key)*/
+
+
+CREATE DATABASE schoolreg;
+
+CREATE TABLE `CLASS` (
+`class_id` INT(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY, 
+`name` VARCHAR (255) NOT NULL,
+`begin_date`  DATETIME,
+`ending_date` DATETIME
+);
+
+CREATE TABLE `STUDENT` (
+`student_id` INT(10) unsigned NOT NULL AUTO_INCREMENT PRIMARY KEY, 
+`name` VARCHAR (255) NOT NULL,
+`email` VARCHAR(255) NOT NULL,
+`Phone` NULL,
+FOREIGN KEY `class_id` INT (10)
+);
+
+--Create an index on the name column of the student table.
+ALTER TABLE student ADD INDEX idx_name (name);
+--Or 
+CREATE INDEX idx_name
+ON student (name);
+/*Adding  a new column to the class table named status which can only have the following values: not-started, ongoing, finished (hint: enumerations).*/
+ALTER TABLE class 
+ADD status 
+Values ENUM(‘not started’, ‘ongoing’, ‘finished’);
+
```

Comment:
When you add the status column what would happen to the existing rows? Which value would status get.

You need to define a default value for this.